src/main/webapp/directives/d3Annotation.js

```diff
@@ -1498,6 +1498,18 @@ angular
                                     // Determine the edges of the path of the link;
                                     // if the position of one of the boxes is not yet known, estimate it
                                     // by using the first word of the corresponding annotation
+                                    if (sourceBox.x === undefined && sourceBox.y === undefined && sourceBox.width === undefined
+                                    && targetBox.x === undefined && targetBox.y === undefined && targetBox.width === undefined) {
```

Comment:
one tiny thing: indent this line
sometimes it's more readable if you write every bool statement in one line